modules/learn/pages/data/distributed-transactions.adoc

```diff
@@ -0,0 +1,315 @@
+    if (bethBalance > transferAmount) {
+	 	andyContent.put("account_balance", andyBalance + transferAmount);
+        txnctx.replace(andy, andyContent);
+    
+        bethContent.put("account_balance", bethBalance - transferAmount);
+        txnctx.replace(beth, bethContent)
+        }
+        else throw new InsufficientFunds();  
+   	// commit transaction - optional, can be omitted
+   	txnctx.commit();
+});
+---- 
+
+The Java example above is a classic example to transfer money between two accounts. 
+Note the use of a lambda function to express the transaction. 
+
+The application supplies the logic for the transaction inside a lambda, including any conditional logic required, and the transactions API takes care of getting the transaction committed. If the transactions API encounters a transient error, such as a temporary conflict with another transaction, then it can rollback what has been done so far and run the lambda again. 
+The application does not have to do these retries and error handling itself.
+
+For more information and examples of using the transactions API, see Java, C++, and C SDK documentation.
+
+NOTE: Use transactions only on documents less than 10 MB in size.
+
+For application-level transactions, create and use transactions through Couchbase SDK APIs. 
+
+Here is another example that combines the usage of key-value and query operations:
+
+[source,java]
+----
+transactions.run((ctx) -> {
+    QueryResult qr = ctx.query(inventory, "UPDATE hotel SET price = $1 WHERE name = $2",
+    TransactionQueryOptions.queryOptions()
+    .parameters(JsonArray.from("from £89", "Glasgow Grand Central")));
+    assert(qr.metaData().metrics().get().mutationCount() == 1);
+});
+----
+
+For more information on distributed transactions through the SDK APIs, see: 
+
+* xref:java-sdk:howtos:distributed-acid-transactions-from-the-sdk.adoc[Java SDK]
+* xref:dotnet-sdk:howtos:distributed-acid-transactions-from-the-sdk.adoc[.NET SDK]
+* xref:cxx-txns:distributed-acid-transactions-from-the-sdk.adoc[C++ API]
+
+
+For use-cases which need to run ad-hoc data changes, you can directly use transactional constructs in N1QL. This can be accomplished using cbq, Query Workbench, CLI, or REST API in Couchbase Server, or through SDKs. 
+
+[source,n1ql]
+----
+START TRANSACTION;
+UPDATE CUSTOMER SET C_BALANCE = C_BALANCE - 1000 WHERE C_FIRST="Beth";
+UPDATE CUSTOMER SET C_BALANCE = C_BALANCE + 1000 WHERE C_FIRST=“Andy”;
+COMMIT ;
+----
+
+For more information on using Query statements in transactions, see xref:n1ql:n1ql-language-reference/transactions.adoc[].
+
+NOTE: Take a look at the https://transactions.couchbase.com[Query Transaction Simulator] which demonstrates how query statements work in transactions.
+
+== Structure of a Transaction
+
+Every transaction has a beginning and a commit or a rollback at the end. 
+
+Every transaction consists of one or more KV operations and one or more query statements. 
```

Comment:
Is that 'and' correct? A transaction could have zero Query statements?

Reply:
How about: Every transaction consists of one or more KV operations, and optionally one or more query statements.